src/Build/Graph/ProjectGraph.cs

```diff
@@ -54,5 +59,139 @@ public ProjectGraph(string entryProjectFile)
         /// Get an unordered collection of all project nodes in the graph.
         /// </summary>
         public IReadOnlyCollection<ProjectGraphNode> ProjectNodes => _projectNodes;
+
+        /// <summary>
+        /// Gets the target list for every node, given a particular target list for the entry project.
+        /// </summary>
+        /// <param name="entryProjectTargets">The target list for the entry project. May be null or empty, in which case the project's default targets will be used.</param>
+        /// <returns>A dictionary containing the target list for each node.</returns>
+        public IReadOnlyDictionary<ProjectGraphNode, string[]> GetTargetLists(string[] entryProjectTargets)
```

Comment:
Should the target list be a readonly collection instead of mutable array?

Reply:
It's an array since the `BuildRequestData` requires an array. Since this is a public API though, perhaps you're right and the conversion to an array should happen when constructing that `BuildRequestData`.